lib/board.rb

```diff
@@ -1,33 +1,63 @@
 class Board 
-	attr_accessor :matrix
-	def initialize
-		@matrix = [nil, nil, nil, nil, nil, nil, nil, nil, nil]
+	attr_accessor :matrix, :win_combinations, :dim
+	def initialize(dim)
+		@matrix = Array.new(dim**2)
+		@win_combinations = init_combin(dim)
+		@dim = dim
+	end
+  
+  	def init_combin(dim)
+  		arr = []
+        arr  += Array.new(dim ){|l| Array.new(dim){|i| dim * l + i }}  
+        arr  += Array.new(dim){|l| Array.new(dim){|i|  l + (i*dim)}}
+        arr  += Array.new(1){|l| Array.new(dim){|i|  i+ (i*dim)}}
+        arr  += Array.new(1){|l| Array.new(dim){|i|  i*(dim-1) + (dim-1)}}
+        return arr                      
+    end
```

Comment:
In general, avoid naming variables with only one character. Find the balance between being brief and explaining what a variable does/stores